gapic/utils/options.py

```diff
@@ -146,6 +146,17 @@ def tweak_path(p):
         # Build the options instance.
         sample_paths = opts.pop('samples', [])
 
+        # autogen-snippets is True by default, so make sure users can disable
+        # by passing `autogen-snippets=false`
+        autogen_snippets = opts.pop(
+            "autogen-snippets", ["True"])[0] in ("True", "true", "T", "t", "TRUE")
+
+        # NOTE: Snippets are not currently correct for the alternative (Ads) templates
+        # so always disable snippetgen in that case
+        # https://github.com/googleapis/gapic-generator-python/issues/1052
+        if opts.get("old-naming"):
+            autogen_snippets = False
```

Comment:
Nit: can you lift the parsing of `old_naming` to right before here and then just reuse the value?